### DISCUSSION TITLE
Why PatchCore validation step takes so much time?

### DISCUSSION THREAD

**Discussion by Yiiipu** [open-edge-platform/anomalib#1763] [Q&A] [answered]
I assume in the training step, the features are extracted and the memory bank is formed, and in the validation step, the validation set is used to find a good threshold. Do I comprehend it correctly? If so, why the validation step of PatchCore could take significantly more time than training step (e.g. 2-4 hours more sometimes, depends on the number of data in the training set)?

**Comment by abc-125** [comment, marked as answer]
I think it can be due to coreset subsampling, which happens after feature extraction at the end of training. It usually takes a lot of time, depending on the training set size.

**Reply by Yiiipu** [reply]
Thank you! So the time consuming coreset sampling actually happens during the training step, and in the validation it simply identifies a threshold? The console messages are misleading because it prints the validation message before coreset indices selection message.

**Reply by abc-125** [reply]
> So the time consuming coreset sampling actually happens during the training step, and in the validation it simply identifies a threshold?

Yes, this should be correct. You're welcome!
